### PR TITLE
Update docs to document new optional status of logback dependency

### PR DESCRIPTION
**Pull request by dgault**
This PR is a follow up to the logback dependency work as per the recommendation inhttps://github.com/ome/bioformats/pull/3943#issuecomment-1422963707

It updates the Bio-Formats docs to make it clearer for users that a logging dependency must be explicitly declared.

### CODE REVIEW

File: sphinx/developers/logging.rst [hunk 1]
```diff
@@ -6,14 +6,22 @@ Logging frameworks
...
-underlying logging frameworks are currently supported by Bio-Formats:
+underlying logging frameworks are currently supported by Bio-Formats. Bio-Formats 
+does not impose any specific SLF4J binding on downstream consumers as per the 
+recommended `SLF4J best practices <https://www.slf4j.org/manual.html#libraries>`_.
+Some recommended bindings are as follows:
 
 - `logback <http://logback.qos.ch/>`_ is the recommended framework and
-  natively implements the SL4J API,
+  natively implements the SL4J API. A logback version of 1.2.x is recommended as 
+  Bio-Formats has a hard dependency for sl4j-api 1.7.30. This means that logbook
```
Comment (sbesson) [inline sphinx/developers/logging.rst:16]:
```suggestion
  Bio-Formats has a hard dependency for sl4j-api 1.7.30. This means that logback 
```

File: sphinx/developers/logging.rst [hunk 2]
```diff
@@ -6,14 +6,22 @@ Logging frameworks
...
+underlying logging frameworks are currently supported by Bio-Formats. Bio-Formats 
+does not impose any specific SLF4J binding on downstream consumers as per the 
+recommended `SLF4J best practices <https://www.slf4j.org/manual.html#libraries>`_.
+Some recommended bindings are as follows:
 
 - `logback <http://logback.qos.ch/>`_ is the recommended framework and
-  natively implements the SL4J API,
+  natively implements the SL4J API. A logback version of 1.2.x is recommended as 
+  Bio-Formats has a hard dependency for sl4j-api 1.7.30. This means that logbook 
+  versions 1.3.x are not yet supported in Bio-Formats as they rely on sl4j-api 2.x.
```
Comment (sbesson) [inline sphinx/developers/logging.rst:17]: Maybe version 1.3.x and higher. A consuming library which does not need to support JDK8 might want to use `logback 1.4.x` but will also need to override `slf4j-api`

File: sphinx/developers/java-library.rst [hunk 3]
```diff
@@ -10,6 +10,9 @@ the `OME Artifactory <http://artifacts.openmicroscopy.org/artifactory>`_.  The
 snippets for inclusion into your Gradle, Maven or Ivy project, which will
 enable the use of this repository.
 
+As of Bio-Formats 6.12.0 consumers will have to directly declare a dependency for
```
Comment (sbesson) [inline sphinx/developers/java-library.rst:13]: Only thought is that we might want to use markup to make this a bit more prominent as an action might be required.

Possibly the [versionchanged](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionchanged) directive could be used here?

File: sphinx/developers/java-library.rst [hunk 4]
```diff
@@ -28,8 +31,8 @@ Alternatively Bio-Formats can be used by including its component jar files.
 You can :downloads:`download formats-gpl.jar <artifacts/formats-gpl.jar>` to
 use it as a library. Just add :file:`formats-gpl.jar` to your :envvar:`CLASSPATH` or
 build path. You will also need :file:`ome-common.jar` for common I/O functions,
-:file:`ome-xml.jar` for metadata standardization, and
-`SLF4J <http://www.slf4j.org/>`_ for :doc:`logging`.
+:file:`ome-xml.jar` for metadata standardization, and a
+`SLF4J <http://www.slf4j.org/>`_ binding for :doc:`logging`.
 
 Dependencies
```
Comment (sbesson) [inline sphinx/developers/java-library.rst:37]: Should the table below indicate that logback is an optional dependency ?

**Review by sbesson — CHANGES_REQUESTED** [review]
A few comments and suggestions but this looks good to me. We might want to add a link to either page in the version history page so that developers can be directed to the instructed about plugging in their binding.

File: sphinx/developers/logging.rst [hunk 5]
```diff
@@ -6,14 +6,22 @@ Logging frameworks
...
 facade and needs to be bound to a logging framework at deployment time. Two
-underlying logging frameworks are currently supported by Bio-Formats:
+underlying logging frameworks are currently supported by Bio-Formats. Bio-Formats 
+does not impose any specific SLF4J binding on downstream consumers as per the 
+recommended `SLF4J best practices <https://www.slf4j.org/manual.html#libraries>`_.
+Some recommended bindings are as follows:
 
 - `logback <http://logback.qos.ch/>`_ is the recommended framework and
-  natively implements the SL4J API,
+  natively implements the SL4J API. A logback version of 1.2.x is recommended as
```
Comment (melissalinkert) [inline sphinx/developers/logging.rst:15]:
```suggestion
  natively implements the SLF4J API. A logback version of 1.2.x is recommended as 
```

File: sphinx/developers/logging.rst (hunk 1 above)
Comment (melissalinkert) [reply sphinx/developers/logging.rst:16]:
```suggestion
  Bio-Formats has a hard dependency for slf4j-api 1.7.30. This means that logback
```

File: sphinx/developers/logging.rst [hunk 6]
```diff
@@ -6,14 +6,22 @@ Logging frameworks
...
-  natively implements the SL4J API,
+  natively implements the SL4J API. A logback version of 1.2.x is recommended as 
+  Bio-Formats has a hard dependency for sl4j-api 1.7.30. This means that logbook 
+  versions 1.3.x are not yet supported in Bio-Formats as they rely on sl4j-api 2.x. 
 - `log4j <http://logging.apache.org/log4j>`_ is the other logging framework
   supported by Bio-Formats and is primarily used in the
   :doc:`MATLAB environment <matlab-dev>`.
 
+Examples of declaring these dependencies using Maven or Gradle are given in the 
+https://github.com/ome/bio-formats-examples repository.
```
Comment (melissalinkert) [inline sphinx/developers/logging.rst:23]:
```suggestion
`Bio-Formats examples repository <https://github.com/ome/bio-formats-examples>`_.
```

**Review by melissalinkert — CHANGES_REQUESTED** [review]

File: sphinx/developers/java-library.rst (hunk 4 above)
Comment (dgault) [reply sphinx/developers/java-library.rst:37]: Yeah, and a lot of the versions are out of date, I will update the table

**Comment by dgault** [comment]
Closing this and will reopen as a clean PR as my branch has gotten out of synch

**Pull request closed without merging**